web/blog/mobilizing-zeek-logs/index.md

```diff
@@ -0,0 +1,212 @@
+schemas can change based on configuration. With `zeek-cut`, it's at least a bit
+more robust.
+
+With Tenzir, it's as easy to process Zeek TSV logs using the native
+[`zeek-tsv`](/next/formats/zeek-tsv) parser:
+
+```bash
+cat *.log | tenzir 'read zeek-tsv | select id.orig_h, id.resp_h'
+```
+
+Tenzir takes care of parsing the type information properly and keeps IP
+addresses and timestamps as nativedata types. You can also see in the examples
+that Tenzir handles multiple concatenated TSV logs of different schemas as you'd
+expect.
+
+Now that Zeek logs are flowing, you can do a lot more than selecting specific
+columns. Check out the [reshaping guide](/next/user-guides/reshape-data) for
+filtering rows, performing aggregations, or routing them elsewhere.
+
+### Demultiplexed NDJSON
+
+Zeek can also render logs as JSON by setting
+[`LogAscii::use_json=T`](https://docs.zeek.org/en/master/frameworks/logging.html):
+
+```bash
+zeek -r trace.pcap LogAscii::use_json=T
+```
+
+As with TSV, this generates one file per log type containing the NDJSON records.
+Here are the same three entries from above in NDJSON:
+
+```json
+{"ts":1258531221.486539,"uid":"C8b0xF1gjm7rOZXemg","id.orig_h":"192.168.1.102","id.orig_p":68,"id.resp_h":"192.168.1.1","id.resp_p":67,"proto":"udp","service":"dhcp","duration":0.1638200283050537,"orig_bytes":301,"resp_bytes":300,"conn_state":"SF","missed_bytes":0,"history":"Dd","orig_pkts":1,"orig_ip_bytes":329,"resp_pkts":1,"resp_ip_bytes":328}
+{"ts":1258531680.237254,"uid":"CMsxKW3uTZ3tSLsN0g","id.orig_h":"192.168.1.103","id.orig_p":137,"id.resp_h":"192.168.1.255","id.resp_p":137,"proto":"udp","service":"dns","duration":3.780125141143799,"orig_bytes":350,"resp_bytes":0,"conn_state":"S0","missed_bytes":0,"history":"D","orig_pkts":7,"orig_ip_bytes":546,"resp_pkts":0,"resp_ip_bytes":0}
+{"ts":1258531693.816224,"uid":"CtTLWk44k1eOCdSjBf","id.orig_h":"192.168.1.102","id.orig_p":137,"id.resp_h":"192.168.1.255","id.resp_p":137,"proto":"udp","service":"dns","duration":3.7486469745635986,"orig_bytes":350,"resp_bytes":0,"conn_state":"S0","missed_bytes":0,"history":"D","orig_pkts":7,"orig_ip_bytes":546,"resp_pkts":0,"resp_ip_bytes":0}
+```
+
+We call this format "multiplexed" because the stream of logs is now spread over
+one file per type. Use the regular [`json`](/next/formats/json) parser to
+mobilize these log files with Tenzir:
+
+```bash
+cat conn.log | tenzir 'read json --schema "zeek.conn" | taste 1'
+```
+
+Note how we pass the type `zeek.conn` to the JSON parser. This sets the name of
+the schema explicitly.
+
+### Multiplexed NDJSON
+
+The above one-file-per-log format is not conducive to stream processing. Most
+analytics tools treat ingestion as a continuous process. Enter [JSON streaming
+logs](https://github.com/corelight/json-streaming-logs). Instead of writing one
+log per file and a custom format, this package makes Zeek produce a single
+stream of NDJSON.
```

Comment:
This doesn't sound right actually. The package still produces one JSON file per log stream, it just changes the structure to include `_path` and `_write_ts` so that all these files can be aggregated easily by the consumer.

Reply:
Thanks for pointing this out. I've tweaked the wording accordingly.